lib/services/metadata.js

```diff
@@ -224,11 +224,21 @@ function putMeta(uri, data) {
  *
  * @param {String} uri
  * @param {Object} data
+ * @param {Object} locals
  * @returns {Promise}
  */
-function patchMeta(uri, data) {
+function patchMeta(uri = '', data = {}, locals = {}) {
   const id = replaceVersion(uri.replace('/meta', ''));
 
+  if (data.shouldPatchArchive) {
+    const { user } = locals,
+      action = data.archived ? 'archive' : 'unarchive',
+      historyEntry = { action, timestamp: new Date().toISOString(), users: [userOrRobot(user)] };
```

Comment:
You could have the logic to create a historyEntry separately. This will improve testing.